File: OOP/polymorphism.py


class Truck:
        def __init__(self, brand, model):
                self.brand = brand
                self.moddel = model


        def move(self):
                print("Moves in the road")



class Ship: 
        def __init__(self, brand, model):
                self.brand = brand
                self.model = model

        def move(self):
                print("Moves in the water")



class Plane:
        def __init__(self, brand, model):
                self.brand = brand
                self.model = model

        def move(self):
                print("Moves in the air")

track = Truck("T", "t")
track.move()


# ship = Ship("S", "s")
# ship.move()


# plane = Plane("P", 'p')
# plane.move()

File: OOP/polymorphism.cpp
#include<bits/stdc++.h>
using namespace std;

class Vehicle{
        public:
        void move(){
                cout << "Can move in roads or water or air" << endl;
        }
};



class Truck : public Vehicle{
        public:
        void move(){
                cout << "Moves in the road" << endl;
        }
};


class Ship : public Vehicle{
        public:
        void move (){
                cout << "Moves in the water" << endl;
        }
};



int main(){
        Vehicle vehicle;
        Truck truck;
        Ship ship;


         vehicle.move();
        truck.move();
        ship.move();

        return 0;
}